extractor/filesystem/language/cpp/conanlock/extractor.go

```diff
@@ -0,0 +1,240 @@
+// Package conanlock extracts conan.lock files.
+package conanlock
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/osv"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+type conanReference struct {
```

Comment:
It looks like we parse this but don't actually store most of the fields in the inventory. Can we introduce a conanlock-specific metadata where we store these?

Reply:
We probably can? Though I'm not very familiar with this format, maybe we should wait till there is a need for it before adding the metadata, it doesn't seem too hard to add afterwards.

Reply:
Okay, in that case though we should remove all these fields from the struct except Name/Version since we aren't using them.

Reply:
Also the fields from conanGraphNode and conanLockFile, e.g. we don't do anything with Options or ProfileHost so we don't need those in the json struct.

Reply:
I removed the extra fields in graphNode and Lockfile. I'm not sure about removing them in conanReference though, as you get those fields as part of the parsing anyway. (See parseConanRenference).

Reply:
Ah right, I guess those kinda need to be parsed anyways (or at least it helps readability if we parse them). Should be fine to leave as is then.